### ISSUE TITLE
An unexpected server error occurred. If this keeps happening, please contact a site administrator.

### ISSUE THREAD

**Issue by starbock** [githubjeka/tracker-issues#42]
**Tracker-Issues:** Latest version: 0.2.3 (installed from marketplace)
**HumHub Community Edition:** Version: 1.7.1
**PHP:** 7.4

**_To Reproduce:_**

- Click "Documents" from top navbar
- Click "+ New Document" button

**_Result_**

- At the bottom of the page I see the "An unexpected server error occurred. If this keeps happening, please contact a site administrator." error message
- The bottom can be opened to reveal this trace information:
```
{
    "url": "/index.php?r=tracker-issues%2Fdocument%2Fcreate",
    "status": 500,
    "response": "<pre>Exception &#039;Error&#039; with message &#039;Class &#039;humhub\\widgets\\RichtextField&#039; not found&#039; \n\nin /home/public_html/protected/modules/tracker_issue/views/document/_form.php:81\n\nStack trace:\n#0 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(348): require()\n#1 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(257): yii\\base\\View-&gt;renderPhpFile(&#039;/home/...",
    "textStatus": "error",
    "xhr": {
        "readyState": 4,
        "responseText": "<pre>Exception &#039;Error&#039; with message &#039;Class &#039;humhub\\widgets\\RichtextField&#039; not found&#039; \n\nin /home/public_html/protected/modules/tracker_issue/views/document/_form.php:81\n\nStack trace:\n#0 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(348): require()\n#1 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(257): yii\\base\\View-&gt;renderPhpFile(&#039;/home/...&#039;, Array)\n#2 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(156): yii\\base\\View-&gt;renderFile(&#039;/home/...&#039;, Array, NULL)\n#3 /home/public_html/protected/modules/tracker_issue/views/document/create.php(21): yii\\base\\View-&gt;render(&#039;_form&#039;, Array)\n#4 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(348): require(&#039;/home/...&#039;)\n#5 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(257): yii\\base\\View-&gt;renderPhpFile(&#039;/home/...&#039;, Array)\n#6 /home/public_html/protected/humhub/modules/ui/view/components/View.php(180): yii\\base\\View-&gt;renderFile(&#039;/home/...&#039;, Array, Object(tracker\\controllers\\DocumentController))\n#7 /home/public_html/protected/vendor/yiisoft/yii2/web/Controller.php(51): humhub\\modules\\ui\\view\\components\\View-&gt;renderAjax(&#039;create&#039;, Array, Object(tracker\\controllers\\DocumentController))\n#8 /home/public_html/protected/modules/tracker_issue/controllers/DocumentController.php(104): yii\\web\\Controller-&gt;renderAjax(&#039;create&#039;, Array)\n#9 [internal function]: tracker\\controllers\\DocumentController-&gt;actionCreate()\n#10 /home/public_html/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)\n#11 /home/public_html/protected/vendor/yiisoft/yii2/base/Controller.php(180): yii\\base\\InlineAction-&gt;runWithParams(Array)\n#12 /home/public_html/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\\base\\Controller-&gt;runAction(&#039;create&#039;, Array)\n#13 /home/public_html/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\\base\\Module-&gt;runAction(&#039;tracker-issues/...&#039;, Array)\n#14 /home/public_html/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\\web\\Application-&gt;handleRequest(Object(humhub\\components\\Request))\n#15 /home/public_html/index.php(25): yii\\base\\Application-&gt;run()\n#16 {main}</pre>",
        "status": 500,
        "statusText": "error"
    },
    "dataType": "html",
    "html": "<pre>Exception &#039;Error&#039; with message &#039;Class &#039;humhub\\widgets\\RichtextField&#039; not found&#039; \n\nin /home/public_html/protected/modules/tracker_issue/views/document/_form.php:81\n\nStack trace:\n#0 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(348): require()\n#1 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(257): yii\\base\\View-&gt;renderPhpFile(&#039;/home/...",
    "error": {},
    "errorThrown": "",
    "validationError": false
}

```
- Additionally, the following INFO entry is logged for this event in the HumHub logger:

```
$_GET = [ 'r' => 'tracker-issues/document/create' ] 
$_SERVER = [ 
'CONTEXT_DOCUMENT_ROOT' => '/home/public_html/' 
'CONTEXT_PREFIX' => '' 
'DOCUMENT_ROOT' => '/home/public_html/' 
'GATEWAY_INTERFACE' => 'CGI/1.1' 
'H2PUSH' => 'on' 
'H2_PUSH' => 'on' 
'H2_PUSHED' => '' 
'H2_PUSHED_ON' => '' 
'H2_STREAM_ID' => '5' 
'H2_STREAM_TAG' => '583-5' 
'HTTP2' => 'on' 
'HTTPS' => 'on' 
'HTTP_ACCEPT' => 'text/html, */*; q=0.01' 
'HTTP_ACCEPT_ENCODING' => 'gzip, deflate, br' 
'HTTP_ACCEPT_LANGUAGE' => 'en-US,en;q=0.9' 
'HTTP_COOKIE' => 'elggperm=zv0lwhHA1UDqYCDyaXcKqlRrQgvOrobQ; Elgg=2e6bf19be7d6f5b3a30cfcb65194fd9e; ossn_chat_bell=off; language=4559af05e7f1d1830c77c2c7838e46a720cae35617e3f38ae5e9325a5a346eb2a%3A2%3A%7Bi%3A0%3Bs%3A8%3A%22language%22%3Bi%3A1%3Bs%3A5%3A%22en-US%22%3B%7D; _identity=eca1d61d337bebd83e834cc896c20e9959f6313276c42ee7b5e0cdca4a1deb68a%3A2%3A%7Bi%3A0%3Bs%3A9%3A%22_identity%22%3Bi%3A1%3Bs%3A50%3A%22%5B1%2C%22954284d0-afb3-42a6-9627-c033872c2df2%22%2C2592000%5D%22%3B%7D; _csrf=94d534c33e49eb9520b6b1eef33ec0b79aaacdd46b441cfbce3797383e96fd3da%3A2%3A%7Bi%3A0%3Bs%3A5%3A%22_csrf%22%3Bi%3A1%3Bs%3A32%3A%22gAavrNpFAs_kUyiQFfSRGVol7f-i_TXl%22%3B%7D; PHPSESSID=d3a72af7b9ad719e7ecaacf8cdb310c6' 
'HTTP_DNT' => '1' 
'HTTP_HOST' => '' 
'HTTP_REFERER' => 'https://index.php?r=tracker-issues%2Fdocument' 
'HTTP_SEC_FETCH_DEST' => 'empty' 
'HTTP_SEC_FETCH_MODE' => 'cors' 
'HTTP_SEC_FETCH_SITE' => 'same-origin' 
'HTTP_USER_AGENT' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36' 
'HTTP_X_CSRF_TOKEN' => 'wHmY2-HVKrdcqLdpntKrj3Zxw76UyLsa-ZKArjxySvunOPmtk5ta8R3b6ALLq8LeMBeQ7NOe1HbO9K3HYyYSlw==' 
'HTTP_X_HTTPS' => '1' 
'HTTP_X_REQUESTED_WITH' => 'XMLHttpRequest' 
'PATH' => '/bin:/usr/bin' 
'QUERY_STRING' => 'r=tracker-issues%2Fdocument%2Fcreate' 
'REDIRECT_STATUS' => '200' 
'REMOTE_ADDR' => 'xxx.xxx.xxx.xxx' 
'REMOTE_PORT' => '55175' 
'REQUEST_METHOD' => 'GET' 
'REQUEST_SCHEME' => 'https' 
'REQUEST_URI' => '/index.php?r=tracker-issues%2Fdocument%2Fcreate' 
'SCRIPT_FILENAME' => '/home/public_html/index.php' 
'SCRIPT_NAME' => '/index.php' 
'SCRIPT_URI' => 'https://index.php' 
'SCRIPT_URL' => '/index.php' 
'SERVER_ADDR' => 'xxx.xxx.xxx.xxx' 
'SERVER_ADMIN' => 'xxx@xxx.xxx' 
'SERVER_NAME' => '' 
'SERVER_PORT' => '443' 
'SERVER_PROTOCOL' => 'HTTP/2.0' 
'SERVER_SIGNATURE' => '' 
'SERVER_SOFTWARE' => 'Apache' 
'SSL_TLS_SNI' => '' 
'TZ' => 'America/New_York' 
'UNIQUE_ID' => 'X868I6FnEZg-8NiLlIivrwACRxM' 
'PHP_SELF' => '/index.php' 
'REQUEST_TIME_FLOAT' => 1607384099.8529 
'REQUEST_TIME' => 1607384099 
'argv' => [ 0 => 'r=tracker-issues%2Fdocument%2Fcreate' ] 
'argc' => 1 ]
```

- Additionally, the following ERROR entry is logged for this event in the HumHub logger:

```
Error: Class 'humhub\widgets\RichtextField' not found in /home/public_html/protected/modules/tracker_issue/views/document/_form.php:81 
Stack trace: 
#0 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(348): require() 
#1 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(257): yii\base\View->renderPhpFile('/home/...', Array) 
#2 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(156): yii\base\View->renderFile('/home/...', Array, NULL) 
#3 /home/public_html/protected/modules/tracker_issue/views/document/create.php(21): yii\base\View->render('_form', Array) 
#4 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(348): require('/home/...') 
#5 /home/public_html/protected/vendor/yiisoft/yii2/base/View.php(257): yii\base\View->renderPhpFile('/home/...', Array) 
#6 /home/public_html/protected/humhub/modules/ui/view/components/View.php(180): yii\base\View->renderFile('/home/...', Array, Object(tracker\controllers\DocumentController)) 
#7 /home/public_html/protected/vendor/yiisoft/yii2/web/Controller.php(51): humhub\modules\ui\view\components\View->renderAjax('create', Array, Object(tracker\controllers\DocumentController)) 
#8 /home/public_html/protected/modules/tracker_issue/controllers/DocumentController.php(104): yii\web\Controller->renderAjax('create', Array) 
#9 [internal function]: tracker\controllers\DocumentController->actionCreate() 
#10 /home/public_html/protected/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array) 
#11 /home/public_html/protected/vendor/yiisoft/yii2/base/Controller.php(180): yii\base\InlineAction->runWithParams(Array) 
#12 /home/public_html/protected/vendor/yiisoft/yii2/base/Module.php(528): yii\base\Controller->runAction('create', Array) 
#13 /home/public_html/protected/vendor/yiisoft/yii2/web/Application.php(103): yii\base\Module->runAction('tracker-issues/...', Array) 
#14 /home/public_html/protected/vendor/yiisoft/yii2/base/Application.php(386): yii\web\Application->handleRequest(Object(humhub\components\Request)) 
#15 /home/public_html/index.php(25): yii\base\Application->run() 
#16 {main}

```

**Comment by ArchBlood** [comment]
Seems to be fine on the repo
https://github.com/githubjeka/tracker-issues/blob/3f770366e9f80b7b2a6eeb2264a0f791cba34adf/views/document/_form.php#L92-L99

**Comment by starbock** [comment]
I believe that you checked v0.3.0 (latest)?  However, marketplace is installing v0.2.3.

**Comment by githubjeka** [comment]
https://github.com/githubjeka/tracker-issues/issues/41#issuecomment-740383500

**Comment by githubjeka** [comment]
Fixed in v0.3.0
The module is available in the market.
